core/src/main/java/org/stellar/anchor/sep31/Sep31Service.java

```diff
@@ -151,6 +151,55 @@ void updateAmounts() throws AnchorException {
     Sep31Transaction txn = Context.get().getTransaction();
     Sep31PostTransactionRequest request = Context.get().getRequest();
     Amount feeResponse = Context.get().getFee();
+    boolean withoutQuote = request.getQuoteId() == null;
+
+    AssetInfo reqAsset = assetService.getAsset(request.getAssetCode(), request.getAssetIssuer());
+    BigDecimal reqAmount = decimal(request.getAmount(), reqAsset);
+    BigDecimal fee = decimal(feeResponse.getAmount(), reqAsset);
+    String feeAsset = String.valueOf(feeResponse.getAsset());
+
+    BigDecimal amountIn;
+    String amountInAsset = reqAsset.getCode();
+    BigDecimal amountOut;
+    String amountOutAsset;
+
+    if (withoutQuote) {
+      boolean strictSend = sep31Config.getPaymentType() == STRICT_SEND;
+      // Without quote.
+      amountOutAsset = amountInAsset;
+      if (strictSend) {
+        // amount_in = req.amount
+        // amount_out = amount_in - amount fee
+        amountIn = reqAmount;
+        amountOut = amountIn.subtract(fee);
+      } else {
+        // amount_in = req.amount + fee
+        // amount_out = req.amount
+        amountIn = reqAmount.add(fee);
+        amountOut = reqAmount;
+      }
+    } else {
+      // With quote
+      Sep38Quote quote = sep38QuoteStore.findByQuoteId(request.getQuoteId());
+      txn.setQuoteId(request.getQuoteId());
+      amountOutAsset = quote.getBuyAsset();
+      amountIn = decimal(quote.getSellAmount(), reqAsset);
+      amountOut = decimal(quote.getBuyAmount(), reqAsset);
+    }
+
+    // Update transactions
+    txn.setAmountFee(String.valueOf(fee));
+    txn.setAmountFeeAsset(feeAsset);
```

Comment:
What to do if the quote price details are complex and contain multiple assets? You mentioned to let it empty?

Reply:
Cool! Good catch! 

Fixed.